pkg/vpc/vpc.go

```diff
@@ -219,9 +222,98 @@ func ImportSubnetsFromList(provider api.ClusterProvider, spec *api.ClusterConfig
 	if err != nil {
 		return err
 	}
+
 	return ImportSubnets(provider, spec, topology, subnets)
 }
 
+func ValidateLegacySubnetsForNodeGroups(spec *api.ClusterConfig, provider api.ClusterProvider) error {
+	// If the cluster endpoint is reachable from the VPC we don't need to check the public subnets
+	if spec.HasPrivateEndpointAccess() {
+		return nil
+	}
+
+	subnetsToValidate := sets.NewString()
+	for _, ng := range spec.NodeGroups {
+		if ng.PrivateNetworking {
+			continue
+		}
+
+		if len(ng.AvailabilityZones) > 0 {
+			// Check only the public subnets that this ng has
+			subnetIDs, err := SelectPublicNodeGroupSubnets(ng.AvailabilityZones, spec)
+			if err != nil {
+				return err
+			}
+			subnetsToValidate.Insert(subnetIDs...)
+		} else {
+			// This ng doesn't have AZs defined so we need to check all public subnets
+			for _, subnet := range spec.VPC.Subnets.Public {
+				subnetsToValidate.Insert(subnet.ID)
+			}
+			break
+		}
+	}
+
+	for _, ng := range spec.ManagedNodeGroups {
+		if len(ng.AvailabilityZones) > 0 {
+			// Check only the public subnets that this ng has
+			subnetIDs, err := SelectPublicNodeGroupSubnets(ng.AvailabilityZones, spec)
+			if err != nil {
+				return err
+			}
+			subnetsToValidate.Insert(subnetIDs...)
+		} else {
+			// This ng doesn't have AZs defined so we need to check all public subnets
+			for _, subnet := range spec.VPC.Subnets.Public {
+				subnetsToValidate.Insert(subnet.ID)
+			}
+			break
+		}
```

Comment:
This block is same as the one for eksctl-managed nodegroups. Might want to extract it into a function defined within `ValidateLegacySubnetsForNodeGroups`  so it has access to `subnetsToValidate`.